theano/gpuarray/extra_ops.py

```diff
@@ -430,8 +430,8 @@ def c_support_code_struct(self, node, nodename):
                         size_t dimBlock[3] = {1, 1, 1};
                         size_t tmp0 = shape[axis]-2;
                         size_t tmp1 = shape[axis]-1;
-                        void* kernel_params[] = {(void*) input->ga.data,
-                                                 (void*) output->ga.data,
+                        void* kernel_params[] = {(void*) ((char *)(input->ga.data) + input->ga.offset),
```

Comment:
@lamblin I don't know exactly what `cuda_callkernel()` does, but it certainly doesn't seem to substitute `data` with `data->devptr + offset` automatically.

At any rate, the point is that we must be careful here to match the declared type of the kernel with the provided argument. If we give `GA_BUFFER`, then we provide a `gpudata*`; If we give `GA_POINTER`, then we provide `devptr+offset`.